### ISSUE TITLE
Related to commands used for finetuning or further learning of pre-trained models

### ISSUE THREAD

**Issue by ohjunee** [ultralytics/yolov5#12361]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

1. I understand pretrained yolov5n.pt that the model has SiLU() as a function of activating all convolution layers. I would like to compare the loss by epoch when actually replacing this activation function with ReLU.  

2. First, This [yolov5n-silu](https://github.com/ultralytics/yolov5#pretrained-checkpoints) (yolov5n.pt).To see what the loss value of pt is from my local server, I would like to learn more about 10 epochs (total 310 epches) to check the results. 

3. In this case, I consider the following command. (The remaining arguments omitted are the same as the default setting.)
In this case, I'd like to know the '--resume' factor affects the training results (loss, mAP).
```
python train.py --resume --epochs 10
```
```
python train.py --epochs 10
```
4. I wonder if there is a difference in the results of the above two commands for fine-tuning by replacing them with ReLU.
5. I'd like to know if there's a code line I can check to track the loss of each layer/block, not by epoch.

**Comment by glenn-jocher** [comment]
@ohjunee ```yaml
# Update train.py with new activation function (e.g. ReLU)
model.model.set_swish(memory_efficient=False)  # Set ReLU activation function
# Train normally for 10 epochs
python train.py --epochs 10
# For tracking layer/block loss, consider custom modification to train.py logging
# E.g. Add loss logging to layers/blocks in 'train.py' via 'model.forward' method
# Please refer to PyTorch documentation or community resources for custom logging guidance
```

**Comment by ohjunee** [comment]
@glenn-jocher Thank you for your quick reply.
1. Which directory script should I refer to for the set_swish() function? 
```
model.model.set_swish(memory_efficient=False) # Set ReLU activation function
```

I tried to access it through the library below, but I couldn't find it.
```
from ultralytics import YOLO
# model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)
model = YOLO('yolov5s.yaml').load('yolov5s.pt')  # build from YAML and transfer weights
model.model.set_swish(memory_efficient=False)
```

many thanks.

**Comment by glenn-jocher** [comment]
@ohjunee The `set_swish()` function is not directly accessible via the Ultralytics YOLO class. To modify the activation function, you will need to directly modify the YOLOv5 GitHub repository code in `models/yolo.py`. You can find the `set_swish()` function within this script. After modifying the activation function, you can proceed with training as usual.

**Comment by ohjunee** [comment]
@glenn-jocher I'm using the code from master branch and I couldn't find the function set_swish in the code in the path below.
[yolov5/models/yolo.py](https://github.com/ultralytics/yolov5/blob/master/models/yolo.py)

Instead, we found a way to change the default_act of common.py 's Conv class to ReLU() and make it learn.
```
class Conv(nn.Module):
    # Standard convolution with args(ch_in, ch_out, kernel, stride, padding, groups, dilation, activation)
    default_act = nn.ReLU()  # default activation nn.SiLU
...
```
Tell me if the route I found is wrong

**Comment by glenn-jocher** [comment]
@ohjunee Ah, my apologies for the confusion. It looks like you're on the right track. Modifying the default activation in `yolov5/models/common.py` should indeed allow you to change the activation function to ReLU for the YOLOv5 model. After making this change, you can proceed with training to compare the loss by epoch. Thank you for bringing this to my attention.

**Comment by ohjunee** [comment]
@glenn-jocher I would like to finetune by reducing the difference in loss by layer, so please explain more about what you said in the example below. 

E.g. Add loss logging to layers/blocks in 'train.py' via 'model.forward' method

**Comment by glenn-jocher** [comment]
@ohjunee Adding loss logging to layers or blocks can be achieved by modifying the `train.py` script in the YOLOv5 repository. Within the `model.forward` method in `yolov5/models/yolo.py`, you can add code to log the loss for specific layers or blocks during each forward pass. This will require modifying the model's forward pass to extract and log the loss values for individual components of the network. Please refer to PyTorch documentation or community resources for specific guidance on logging loss by layer or block within the `model.forward` method.
